gc_tests/run_tests.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2019 King's College London created by the Software Development
+// Team <http://soft-dev.org/>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0>, or the MIT license <LICENSE-MIT
+// or http://opensource.org/licenses/MIT>, or the UPL-1.0 license
+// <http://opensource.org/licenses/UPL> at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use std::{env, path::PathBuf, process::Command};
+
+use lang_tester::LangTester;
+use tempdir::TempDir;
+
+static CRATE_NAME: &'static str = "gcmalloc";
+
+fn main() {
+    let tempdir = TempDir::new("gc_tester").unwrap();
+    LangTester::new()
+        .test_dir("gc_tests/tests")
+        .test_file_filter(|p| p.extension().unwrap().to_str().unwrap() == "rs")
+        .test_extract(|s| {
+            Some(
+                s.lines()
+                    .skip_while(|l| !l.starts_with("//"))
+                    .take_while(|l| l.starts_with("//"))
+                    .map(|l| &l[2..])
+                    .collect::<Vec<_>>()
+                    .join("\n"),
+            )
+        })
+        .test_cmds(move |p| {
+            // We grab the rlibs from `target/<debug | release>/` but in order
+            // for them to exist here, they must have been moved from `deps/`.
+            // Simply running `cargo test` will not do this, instead, we must
+            // ensure that `cargo build` has been run before running the tests.
+            let mut build_libs = Command::new("cargo");
```

Comment:
This will run `cargo build` for every lang test: can't it be run just once, up front?